### PR TITLE
[TASK] improve delete message for tags/fieldcollections/objectbricks/…

### LINKED ISSUE #6259: [Improvement/Epic] More delete confirmations and giving element name about to be deleted
Hello,

Following https://github.com/pimcore/pimcore/issues/6240 which is really a great things, I went through all the Pimcore GUI to check if deleting elements are risky.
I found the followings which could be improved.

There is no confirmation, as soon clicked, as soon lost ; )
- tag & snippet management
- qr code
- targeting rules/groups
- custom reports
- thumbnails

There is a popin confirmation but it doesn't diplay the name of the element which is about to be deleted, which could be prone to error
- tags
- roles / users
- fieldcollection
- objectbricks

Those are a little different as it use the small red `x` icon in extjs grid, but I think they are valuable to a confirm popin:
- keydefinition from storecollections... this could be a pretty hard delete has there is no confirm and maybe data in bdd are lost ?
- groups from storecollections

In more generally there are other part in the GUI with red `x` icon without any confirmation (like static routes, etc.). They are less sensitive but can defenitly break an existing application if clicked by error.
I think maybe a global handler on those red "x" could be added.

This issue could be splitted in smallest issues. It could be left as it and planned later.
But in the end purpose is to achieve a better UI with good confirm message and when possible saying what is about to be deleted.

In our production project we have sometimes editors saying, it doesn't work no more, and we found that something has been deleted. Arh : (( We try to limit permissions for sure, but it is often a human wrong clic and having a good preventing system with message could really imporve this I think.

In the long run, it will prepare Pimcore for touch screen, as wrong click can be really easy this way, and deleting should be confirmed for sure.

Best regards.

### PR DESCRIPTION
**Pull request by dpfahlbusch**
…users and add confirmdialog to thumbnails/customreports

## Changes in this pull request  
Partly Resolves #6259

### CODE REVIEW

**Comment by dpfahlbusch** [comment]
@brusch should be ready for review :)

**Comment by blankse** [comment]
Wouldn't it be better to just create one translation with 2 parameters?

**Comment by blankse** [comment]
Can you use the translation function for the type parameter? So this can also be translated.

**Review by blankse — APPROVED** [review]
LGTM, thank you 👍🏼

**Comment by dpfahlbusch** [comment]
Just added a minor change, because i was working on the menus:
changed the icon of delete fieldcollection (was in an earlier commit today) and delete objectbrick to make it consistent with delete class :)

File: bundles/AdminBundle/Resources/public/js/pimcore/object/classificationstore/collectionsPanel.js [hunk 1]
```diff
@@ -311,15 +315,19 @@ pimcore.object.classificationstore.collectionsPanel = Class.create({
...
-                        Ext.Ajax.request({
-                            url: Routing.generate('pimcore_admin_dataobject_classificationstore_deletecollection'),
-                            method: 'DELETE',
-                            params: {
-                                id: id
-                            },
-                            success: function (response) {
-                                this.collectionsStore.reload();
-                            }.bind(this)});
+                        Ext.Msg.confirm(t('delete'), sprintf(t('delete_message_advanced'), t('classificationstore_collection'), data.data.text), function(btn) {
```
Comment (aryaantony92) [inline bundles/AdminBundle/Resources/public/js/pimcore/object/classificationstore/collectionsPanel.js:318]: Ext.Msg.confirm(t('delete'), sprintf(t('delete_message_advanced'), t('classificationstore_collection'), data.data.text), function(btn) {

Comment (aryaantony92) [inline bundles/AdminBundle/Resources/public/js/pimcore/object/classificationstore/collectionsPanel.js]:
```suggestion
                        Ext.Msg.confirm(t('delete'), sprintf(t('delete_message_advanced'), t('classificationstore_collection'), data.data.name), function(btn) {
```

File: bundles/AdminBundle/Resources/public/js/pimcore/object/classificationstore/groupsPanel.js [hunk 2]
```diff
@@ -145,16 +145,20 @@ pimcore.object.classificationstore.groupsPanel = Class.create({
...
-                            url: Routing.generate('pimcore_admin_dataobject_classificationstore_deleterelation'),
-                            method: 'DELETE',
-                            params: {
-                                keyId: keyId,
-                                groupId: groupId
-                            },
-                            success: function (response) {
-                                this.relationsStore.reload();
-                            }.bind(this)});
+                        Ext.Msg.confirm(t('delete'), sprintf(t('delete_message_advanced'), t('classificationstore_relation'), data.data.text), function(btn) {
```
Comment (aryaantony92) [inline bundles/AdminBundle/Resources/public/js/pimcore/object/classificationstore/groupsPanel.js]:
```suggestion
                        Ext.Msg.confirm(t('delete'), sprintf(t('delete_message_advanced'), t('classificationstore_relation'), data.data.keyName), function(btn) {
```

File: bundles/AdminBundle/Resources/public/js/pimcore/object/classificationstore/groupsPanel.js [hunk 3]
```diff
@@ -316,15 +320,19 @@ pimcore.object.classificationstore.groupsPanel = Class.create({
...
-                        Ext.Ajax.request({
-                            url: Routing.generate('pimcore_admin_dataobject_classificationstore_deletegroup'),
-                            method: 'DELETE',
-                            params: {
-                                id: id
-                            },
-                            success: function (response) {
-                                this.groupsStore.reload();
-                            }.bind(this)});
+                        Ext.Msg.confirm(t('delete'), sprintf(t('delete_message_advanced'), t('classificationstore_group'), data.data.text), function(btn) {
```
Comment (aryaantony92) [inline bundles/AdminBundle/Resources/public/js/pimcore/object/classificationstore/groupsPanel.js]:
```suggestion
                        Ext.Msg.confirm(t('delete'), sprintf(t('delete_message_advanced'), t('classificationstore_group'), data.data.name), function(btn) {
```

File: bundles/AdminBundle/Resources/public/js/pimcore/settings/staticroutes.js [hunk 4]
```diff
@@ -185,7 +185,11 @@ pimcore.settings.staticroutes = Class.create({
                         if (!data.data.writeable) {
                             return;
                         }
-                        grid.getStore().removeAt(rowIndex);
+                        Ext.Msg.confirm(t('delete'), sprintf(t('delete_message_advanced'), t('staticroute'), data.data.text), function (btn) {
```
Comment (aryaantony92) [inline bundles/AdminBundle/Resources/public/js/pimcore/settings/staticroutes.js]:
```suggestion
                        Ext.Msg.confirm(t('delete'), sprintf(t('delete_message_advanced'), t('staticroute'), data.data.name), function (btn) {
```

**Review by aryaantony92 — CHANGES_REQUESTED** [review]

File: bundles/AdminBundle/Resources/public/js/pimcore/object/classificationstore/propertiesPanel.js [hunk 5]
```diff
@@ -198,15 +198,19 @@ pimcore.object.classificationstore.propertiespanel = Class.create({
...
-                        Ext.Ajax.request({
-                            url: Routing.generate('pimcore_admin_dataobject_classificationstore_deleteproperty'),
-                            method: 'DELETE',
-                            params: {
-                                id: id
-                            },
-                            success: function (response) {
-                                this.store.reload();
-                            }.bind(this)});
+                        Ext.Msg.confirm(t('delete'), sprintf(t('delete_message_advanced'), t('classificationstore_property'), data.data.text), function(btn) {
```
Comment (aryaantony92) [inline bundles/AdminBundle/Resources/public/js/pimcore/object/classificationstore/propertiesPanel.js]:
```suggestion
                        Ext.Msg.confirm(t('delete'), sprintf(t('delete_message_advanced'), t('classificationstore_property'), data.data.name), function(btn) {
```

**Review by aryaantony92 — CHANGES_REQUESTED** [review]

File: bundles/AdminBundle/Resources/public/js/pimcore/object/classificationstore/collectionsPanel.js [hunk 6]
```diff
@@ -141,16 +141,20 @@ pimcore.object.classificationstore.collectionsPanel = Class.create({
...
-                            url: Routing.generate('pimcore_admin_dataobject_classificationstore_deletecollectionrelation'),
-                            method: 'DELETE',
-                            params: {
-                                colId: colId,
-                                groupId: groupId
-                            },
-                            success: function (response) {
-                                this.relationsStore.reload();
-                            }.bind(this)});
+                        Ext.Msg.confirm(t('delete'), sprintf(t('delete_message_advanced'), t('classificationstore_collection_relation'), data.data.text), function(btn) {
```
Comment (aryaantony92) [inline bundles/AdminBundle/Resources/public/js/pimcore/object/classificationstore/collectionsPanel.js]:
```suggestion
                        Ext.Msg.confirm(t('delete'), sprintf(t('delete_message_advanced'), t('classificationstore_collection_relation'), data.data.groupName), function(btn) {
```

File: bundles/AdminBundle/Resources/public/js/pimcore/settings/targeting/rules/panel.js [hunk 7]
```diff
@@ -192,15 +192,19 @@ pimcore.settings.targeting.rules.panel= Class.create({
...
+                    method: 'DELETE',
+                    params: {
+                        id: record.data.id
+                    },
+                    success: function () {
+                        this.tree.getStore().load();
+                    }.bind(this)
+                });
+            }
         });
```
Comment (aryaantony92) [inline bundles/AdminBundle/Resources/public/js/pimcore/settings/targeting/rules/panel.js]:
```suggestion
       }.bind(this));
```

File: bundles/AdminBundle/Resources/public/js/pimcore/settings/targeting/targetGroups/panel.js [hunk 8]
```diff
@@ -161,17 +161,21 @@ pimcore.settings.targeting.targetGroups.panel= Class.create({
...
+                    success: function () {
+                        this.tree.getStore().load();
 
-                pimcore.globalmanager.get("target_group_store").reload();
-            }.bind(this)
+                        pimcore.globalmanager.get("target_group_store").reload();
+                    }.bind(this)
+                });
+            }
         });
```
Comment (aryaantony92) [inline bundles/AdminBundle/Resources/public/js/pimcore/settings/targeting/targetGroups/panel.js]:
```suggestion
      }.bind(this));
```

**Review by aryaantony92 — CHANGES_REQUESTED** [review]

**Review by aryaantony92 — APPROVED** [review]

**Comment by aryaantony92** [comment]
@dpfahlbusch Thank you

**Pull request merged**